src/Sdk/WebApi/WebApi/BrokerHttpClient.cs

```diff
@@ -111,5 +117,33 @@ public class BrokerHttpClient : RawHttpClientBase
 
             throw new Exception($"Failed to get job message: {result.Error}");
         }
+
+        public async Task<TaskAgentSession> CreateSessionAsync(
+
+           TaskAgentSession session,
+           CancellationToken cancellationToken = default)
+        {
+
+            var requestUri = new Uri(Client.BaseAddress, "session");
+            var requestContent = new ObjectContent<TaskAgentSession>(session, new VssJsonMediaTypeFormatter(true));
+
+            var result = await SendAsync<TaskAgentSession>(
+                new HttpMethod("POST"),
+                requestUri: requestUri,
+                content: requestContent,
+                cancellationToken: cancellationToken);
+
+            if (result.IsSuccess)
+            {
+                return result.Value;
+            }
+
+            if (result.StatusCode == HttpStatusCode.Forbidden)
+            {
+                throw new AccessDeniedException(result.Error);
+            }
```

Comment:
Broker responds with 401s or 400s right now, no 403s: https://github.com/github/actions-broker-listener/blob/eb08f036c89ce6e385e62ddbeae5f84074d7d94c/internal/middleware/jwt_validation_middleware.go#L47-L65

Reply:
Also! Broker responds `409 Conflict` when there's already an active session. https://github.com/github/actions-broker-listener/pull/1031/files#diff-37c615ad33e1d15f935093bd9fc81ccecce06ed47240d27cacce9ced12d79ee3R303

Reply:
👍 Runner already handles the 409 conflict exception